src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/TypeAccessibilityAddUsingCSharpCodeActionResolver.cs

```diff
@@ -0,0 +1,147 @@
+        public TypeAccessibilityAddUsingCSharpCodeActionResolver(
+            ForegroundDispatcher foregroundDispatcher,
+            DocumentResolver documentResolver,
+            ClientNotifierServiceBase languageServer,
+            DocumentVersionCache documentVersionCache)
+            : base(languageServer)
+        {
+            if (foregroundDispatcher is null)
+            {
+                throw new ArgumentNullException(nameof(foregroundDispatcher));
+            }
+
+            if (documentResolver is null)
+            {
+                throw new ArgumentNullException(nameof(documentResolver));
+            }
+
+            if (documentVersionCache is null)
+            {
+                throw new ArgumentNullException(nameof(documentVersionCache));
+            }
+
+            _foregroundDispatcher = foregroundDispatcher;
+            _documentResolver = documentResolver;
+            _documentVersionCache = documentVersionCache;
+        }
+
+        public override string Action => LanguageServerConstants.CodeActions.AddUsing;
+
+        public async override Task<CodeAction> ResolveAsync(
+            CSharpCodeActionParams csharpParams,
+            CodeAction codeAction,
+            CancellationToken cancellationToken)
+        {
+            if (csharpParams is null)
+            {
+                throw new ArgumentNullException(nameof(csharpParams));
+            }
+
+            if (codeAction is null)
+            {
+                throw new ArgumentNullException(nameof(codeAction));
+            }
+
+            var resolvedCodeAction = await ResolveCodeActionWithServerAsync(codeAction, cancellationToken);
+            if (resolvedCodeAction.Edit?.DocumentChanges is null)
+            {
+                // Unable to resolve code action with server, return original code action
+                return codeAction;
+            }
+
+            if (resolvedCodeAction.Edit.DocumentChanges.Count() != 1)
+            {
+                // We don't yet support multi-document code actions, return original code action
+                return codeAction;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var documentChanged = resolvedCodeAction.Edit.DocumentChanges.First();
+            if (!documentChanged.IsTextDocumentEdit)
+            {
+                // Only Text Document Edit changes are supported currently, return original code action
+                return codeAction;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var addUsingTextEdit = documentChanged.TextDocumentEdit.Edits.FirstOrDefault();
+            if (addUsingTextEdit is null)
+            {
+                // No text edit available
+                return codeAction;
+            }
+
+            var @namespace = AddUsingsCodeActionProviderFactory.ExtractNamespaceFromVSCSharpAddUsing(addUsingTextEdit.NewText);
+            if (string.IsNullOrEmpty(@namespace))
```

Comment:
instead of grabbing out a name, then checking for a special value, just make this `TryExtractNamespace`.  And you only proceed if it succeeds.